tasmota/xdrv_39_thermostat.ino

```diff
@@ -2022,8 +2022,16 @@ const char HTTP_THERMOSTAT_HL[]          PROGMEM = "{s}<hr>{m}<hr>{e}";
 
 #endif  // USE_WEBSERVER
 
-void ThermostatShow(uint8_t ctr_output)
+void ThermostatShow(uint8_t ctr_output, bool json)
 {
+  if (json) {
+    ResponseAppend_P(PSTR(",\"Thermostat%i\":{"), ctr_output);
+    ResponseAppend_P(PSTR("%s\"%s\":%i"), "", D_CMND_THERMOSTATMODESET, Thermostat[ctr_output].status.thermostat_mode);
+    ResponseAppend_P(PSTR("%s\"%s\":%f"), ",", D_CMND_TEMPTARGETSET, (Thermostat[ctr_output].temp_target_level / 10.0f));
```

Comment:
Tasmota does not support "%f". It uses too much code space. Instead use "%2_f". (In that case prefix the variable with an "&" too).
Look for example in the code.

Reply:
Fixed it. Thanks.